pywinauto/linux/atspi_element_info.py

```diff
@@ -0,0 +1,232 @@
+    IATSPI
+from ..element_info import ElementInfo
+
+
+class AtspiElementInfo(ElementInfo):
+
+    """Wrapper for window handler"""
+
+    atspi_accessible = AtspiAccessible()
+
+    def __init__(self, handle=None):
+        """Create element by handle (default is root element)"""
+        if handle is None:
+            self._handle = self.atspi_accessible.get_desktop(0)
+        else:
+            self._handle = handle
+
+    def __get_elements(self, root, tree, **kwargs):
+        tree.append(root)
+        for el in root.children(**kwargs):
+            self.__get_elements(el, tree, **kwargs)
+
+    def __hash__(self):
+        """Return a unique hash value based on the element's handle"""
+        return hash(self.handle)
+
+    def __eq__(self, other):
+        """Check if two AtspiElementInfo objects describe the same element"""
+        if not isinstance(other, AtspiElementInfo):
+            return False
+        if self.control_type == "Application" and other.control_type == "Application":
+            return self.process_id == other.process_id
+        return self.rectangle == other.rectangle
+
+    def __ne__(self, other):
+        """Check if two AtspiElementInfo objects describe different elements"""
+        return not (self == other)
+
+    @staticmethod
+    def _get_states_as_string(states):
+        string_states = []
+        for i, state in AtspiStateEnum.items():
+            if states & (1 << i):
+                string_states.append(state)
+        return string_states
+
+    @property
+    def handle(self):
+        """Return the handle of the window"""
+        return self._handle
+
+    @property
+    def name(self):
+        """Return the text of the window"""
+        return self.atspi_accessible.get_name(self._handle, None).decode(encoding='UTF-8')
+
+    @property
+    def control_id(self):
+        """Return the ID of the window"""
+        return self.atspi_accessible.get_role(self._handle, None)
+
+    @property
+    def runtime_id(self):
+        """Return the runtime ID of the element"""
+        return self.atspi_accessible.get_index_in_parent(self._handle, None)
+
+    @property
+    def process_id(self):
+        """Return the ID of process that controls this window"""
+        return self.atspi_accessible.get_process_id(self._handle, None)
+
+    @property
+    def class_name(self):
+        """Return the class name of the element"""
+        role = self.atspi_accessible.get_role_name(self._handle, None)
+        return "".join([part.capitalize() for part in role.decode("utf-8").split()])
+
+    @property
+    def rich_text(self):
+        """Return the text of the element"""
+        return self.name
+
+    @property
+    def control_type(self):
+        """Return the class name of the element"""
+        role_id = self.atspi_accessible.get_role(self._handle, None)
+        return IATSPI().known_control_type_ids[role_id]
```

Comment:
It's worth catching `IndexError` and re-raising it as `NotImplementedError('Unknown role ID has been retrieved: ...')`